### DISCUSSION TITLE
Need Help on consuming SOAP service for Quarkus project

### DISCUSSION THREAD

**Discussion by omkar62** [quarkiverse/quarkus-cxf#438] [Q&A]
Hello,

Currently I have a spring boot project that is integrating with SOAP service which is on another on premise server. I need to convert this project to Quarkus. I am unable to understand how do I change the SOAP integration in Quarkus project.

Currently in springboot, I am using SOAPMessage to create SOAP headers and SOAP body. And for connection with the SOAP service I am using below

import org.springframework.http.ResponseEntity;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;

String requestXML;   //This  includes SOAP headers and SOAP body which is in the form of SOAP Message converted to string
ResponseEntity<String> responseXML;
HttpHeaders headers= new HTTPHeaders();
String endPoint="http://10.167.78.89:9085";  //Intensionally provided dummy url

HttpEntity<String> entity = new HttpEntity<>(requestXML,headers);
responseXML= restTemplate.exchange(endPoint, HttpMethod.POST,entity, string.class);

My understanding is creating soap header and body is a java part and should be same for Quarkus as well. However what is the equivalent for restTemplate in Quarkus that can be used to make a connection with the endpoint url. I want to know how do I make connection with the soap service with a given endpoint as above?

Any help is greatly appreciated. Thanks

**Comment by shumonsharif** [comment]
Hi @omkar62 I don't believe there is a direct equivalent for what you are trying to achieve, at least through this extension.  Having said that, if you can put together a sample Spring Boot project that demonstrates your use case, then I'll be happy to try to work on a Quarkus version of it.

You can also take a look here for the rest-template equivalent in Quarkus:
https://quarkus.io/guides/rest-client

**Reply by omkar62** [reply]
Sure Shumon. Currently project is my official repo. I am working on creating a sample project for your reference. Will share details soon. Thanks

**Comment by omkar62** [comment]
[sample.zip](https://github.com/quarkiverse/quarkus-cxf/files/8679921/sample.zip)

Hi Shumon,

Please find attached sample springboot projected created. Actual project has bunch of request objects but I have created this one to simplify it. 
1. Controller is getting a simple request- name and email fields
2. In SampleHelper class, there are multiple functions to create SOAP header, SOAP body, marshal the request to ServiceReq object
3. The invokeService method in SampleHelper class is connecting with the service using an endpoint and SOAP header & body using restTemplate. This service accepts request only in SOAP format. 
 
In my project, we are using multiple applications to connect to this SOAP service using springboot application. We are planning to use Quarkus and this is the first attempt.

Please provide your suggestions. Thanks for all your help in advance.

**Reply by shumonsharif** [reply]
Hey @omkar62 I took a stab at it, with the goal of minimizing the amount of changes required.  

There was no need to use the quarkus-cxf extension; hope this helps, and good luck with adopting Quarkus!

[sample.zip](https://github.com/quarkiverse/quarkus-cxf/files/8687222/sample.zip)

**Reply by omkar62** [reply]
Thanks Shumon.Let me have a look at this. Also I was able to connect using SOAPConnection class provided by Java. I am testing it out and also will test the code you sent and let you know.
